### PR TITLE
framework/src/media: modify media.

### PR DESCRIPTION
**Pull request by YiHyunJin**
Modify deadlock and error. modify server_linux.

### CODE REVIEW

File: framework/src/media/media_player.c [hunk 1]
```diff
@@ -150,14 +150,12 @@ play_result_t media_stop_play(void)
...
-	PLAY_UNLOCK();
-
-	pthread_join(g_pc.pth, NULL);
-	
-	PLAY_LOCK();
 	g_pc.pth = -1;
+
 	PLAY_UNLOCK();
 
+	pthread_join(g_pc.pth, NULL);
```
Comment (sunghan-chang) [inline framework/src/media/media_player.c]: Definitely wrong. You set ```g_pc.pth``` as -1 at above line. How do you use it in here?

File: framework/src/media/media_player.c [hunk 2]
```diff
@@ -323,7 +321,7 @@ int player_worker(void *args)
 	g_pc.pcmout = NULL;
 
 	PLAY_LOCK();
-	g_pc.state = PLAY_STOPPING;
+	g_pc.state = PLAY_ERROR;
```
Comment (sunghan-chang) [inline framework/src/media/media_player.c]: Why ERROR?

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/media_test/media_test_main.c:34: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/media/media_player.c:246: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

File: apps/examples/media_test/media_test_main.c [hunk 3]
```diff
@@ -160,7 +160,7 @@ int media_test_cb(int argc, char **argv)
 			}
 
 			printf("Socket connecting...\n");
-			fd = connect_socket("192.168.1.213", 8080);
```
[the remark this answers is not in the document]
Comment (YiHyunJin) [reply apps/examples/media_test/media_test_main.c]: file? media_test_main.c? or this line?

File: framework/src/media/media_player.c (hunk 2 above)
Comment (YiHyunJin) [reply framework/src/media/media_player.c]: Later i will edit g_pc.state whole. and if state is not ERROR, restart is not possible.

File: apps/examples/media_test/media_test_main.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsdosa) [reply apps/examples/media_test/media_test_main.c]: this file

File: framework/src/media/media_player.c [hunk 4]
```diff
@@ -245,7 +243,7 @@ int read_data_pcm_write(char *buffer)
 		} else {
 			ret = read(g_pc.fd, buffer + readed, g_pc.buffer_size - readed);
 		}
-		printf("ret = %d\n", ret);
+		printf("ret = %d\n",ret);
```
Comment (jsdosa) [inline framework/src/media/media_player.c]: you should add space.

File: framework/src/media/media_player.c (hunk 1 above)
Comment (jsdosa) [reply framework/src/media/media_player.c]: why do you change that?

File: apps/examples/media_test/server/server_example_linux.c [hunk 5]
```diff
@@ -227,11 +225,12 @@ void *send_thread(void *arg)
...
-
-	fclose(fp);
-
+	if (fp) {
+		fclose(fp);
+		fp = NULL;
+	}
 	printf("Send thread finished.\n");
-
+	g_sc.send_state = SEND_STATE_NONE;
```
Comment (jsdosa) [inline apps/examples/media_test/server/server_example_linux.c]: add lock & unlock

File: framework/src/media/media_player.c (hunk 2 above)
Comment (sunghan-chang) [reply framework/src/media/media_player.c]: Same question. Why ERROR? Do you think starting from ERROR state is correct? Not IDLE? Not STOPPED?

**Comment by sunghan-chang** [comment]
@YiHyunJin I write again. 
```
"modify media" is not enough as commit title. Please write more details.
```

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by YiHyunJin** [comment]
wrong commit, i will re commit

**Pull request closed without merging**